src/sentry/api/endpoints/group_tagkey_values.py

```diff
@@ -51,30 +50,26 @@ def get(self, request, group, key):
         except tagstore.TagKeyNotFound:
             raise ResourceDoesNotExist
 
-        queryset = tagstore.get_group_tag_value_qs(
-            group.project_id, group.id, environment_id, lookup_key)
-
         sort = request.GET.get('sort')
         if sort == 'date':
             order_by = '-last_seen'
-            paginator_cls = DateTimePaginator
         elif sort == 'age':
             order_by = '-first_seen'
-            paginator_cls = DateTimePaginator
         else:
             order_by = '-id'
-            paginator_cls = Paginator
 
         if key == 'user':
             serializer_cls = UserTagValueSerializer(group.project_id)
         else:
             serializer_cls = None
 
+        paginator = tagstore.get_group_tag_value_paginator(
+            group.project_id, group.id, environment_id, lookup_key, order_by=order_by
+        )
+
         return self.paginate(
             request=request,
-            queryset=queryset,
-            order_by=order_by,
-            paginator_cls=paginator_cls,
+            paginator=paginator,
             on_results=lambda results: serialize(
                 map(  # XXX: This is a pretty big abstraction leak
```

Comment:
I think this is probably the right place to fix this? (I think "fixing" means just passing this as a keyword argument inside `get_group_tag_value_paginator` instead?) This was working on the assumption that this was always returning model instances which is no longer the case for all backends.